cmd/resolve-version/main.go

```diff
@@ -40,6 +41,72 @@ type release struct {
 	version  *semver.Version
 }
 
+type matchResult struct {
```

Comment:
The previous `release, error` return type for `matchRelease*` only captured two of the possible cases:

- Search succeeded and there was a result
- Something went wrong

There is a third case that this helps distinguish:

- Search succeeded without a matching result

If there is a more idiomatic way of representing this, I'd love to know